src/shared/module-util.c

```diff
@@ -3,6 +3,43 @@
 #include <errno.h>
 
 #include "module-util.h"
+#include "proc-cmdline.h"
+#include "static-destruct.h"
+#include "strv.h"
+
+static char **arg_proc_cmdline_module_blacklist = NULL;
+static bool blacklist_parsed = false;
+
+STATIC_DESTRUCTOR_REGISTER(arg_proc_cmdline_module_blacklist, strv_freep);
+
+static int blacklist_modules(const char *p) {
+        _cleanup_strv_free_ char **k = NULL;
+
+        k = strv_split(p, ",");
+        if (!k)
+                return log_oom();
+
+        if (strv_extend_strv(&arg_proc_cmdline_module_blacklist, k, true) < 0)
```

Comment:
Then,
```c
static int blacklist_modules(const char *p, char ***blacklist) {
        _cleanup_strv_free_ char **k = NULL;

        assert(p);
        assert(blacklist);

        k = strv_split(p, ",");
        if (!k)
                return -ENOMEM;

        if (strv_extend_strv(blacklist, k, true) < 0)
                return -ENOMEM;

        return 0;
}

static int parse_proc_cmdline_item(const char *key, const char *value, void *data) {
        int r;

        if (proc_cmdline_key_streq(key, "module_blacklist")) {

                if (proc_cmdline_value_missing(key, value))
                        return 0;

                r = blacklist_modules(value, data);
                if (r < 0)
                        return r;
        }

        return 0;
}
```
Note, please do not call `log_oom()`, but please simply return `-ENOMEM`, as these are library functions.